### DISCUSSION TITLE
Way to expand Product-Condition selection ( PS 8.1+)

### DISCUSSION THREAD

**Discussion by darkWolf-PR** [PrestaShop/PrestaShop#34702] [Q&A]
Hi, is there a way to expand the Condition selection in Admin-product? I mean some way, that will support upgrading, without overwriting system files? 

We had this changed(by overwriting system files) in PS 1.6 and now, we are going to upgrade our eshop to PS8 and I´m failing to find a way to extend condition selection in PS 8.1+ and can´t do the eshop upgrade until I have this sorted out - we have about 8 different condition variants instead of default 3 in PS.

Thanks for any help.

**Comment by MatShir** [comment]
One idea would be to change condition as a feature and determine on the product page what feature would correspond to condition 🤔  So the condition could be edited as easy as features

**Reply by darkWolf-PR** [reply]
I can´t switch condition to features, because there are several other systems connected right to the condition field and those would not work without being right inside the Product table, with product ID.

I´ve actually managed to setup a module which changes the condition select field to what I need(based on this [demo](https://github.com/PrestaShop/example-modules/tree/master/demoproductform) ) but after that, I did not found a way, how to change validation against the ValueObject\ProductCondition and so it´s useless... It offers selectable conditions, it even shows the right one selected, but can´t be saved, because it validates against the base 3 (new, used, refurbished) and I just can´t find a way how to modify this.

Demos do not help with this as they are all working with adding new fields and that´s kind of without problems, because you have to validate your new field on your own...

For now, I had to go the wrong way and just rewrite the files in SRC and will have to rewrite them after updates again...

**Reply by kpodemski** [reply]
@darkWolf-PR 

well, I think it is doable but requires much more work, I'll give you some idea on how you could do it:

You have to modify `UpdateProductCommand` which expects `$condition` to be `ProductCondition` value object. 
`UpdateProductCommand` is being used in `PrestaShop\PrestaShop\Adapter\Product\CommandHandler\UpdateProductHandler` a handler class which is available as a service:

```
PrestaShop\PrestaShop\Adapter\Product\CommandHandler\UpdateProductHandler:
    autowire: true
    public: false
    autoconfigure: true
    arguments:
      $productUpdatablePropertyFiller: '@PrestaShop\PrestaShop\Adapter\Product\Update\Filler\ProductFiller'
```

Documentation on how to deal with services decoration is here: https://devdocs.prestashop-project.org/8/modules/concepts/services/#decorate-the-service

I think you could change this handler, so instead on: `UpdateProducCommand`, it'd rely on your own implementation, which extends `UpdateProductCommand`. That's theory, if you are motivated to give it a try please let us know if that works :D
